mitodl/mit-open
# Update docker-compose config to support OS cluster

### What are the relevant tickets?




Part of https://github.com/mitodl/hq/issues/5179

### Description (What does it do?)

This updates the docker compose setup to support running OpenSearch in a 3-node cluster locally.

### How can this be tested?

- You should be able to run `docker compose up` as usual without any additional configuration.
- Test the 3-node cluster:
  - `docker compose stop` if it's running
  - Set `export OPENSEARCH_CLUSTER_TYPE=cluster` in your shell
  - `docker compose up` - you should see 3 opensearch nodes instead of 1
  - Run `manage.py recreate_index --all` as the new cluster uses separate volumes to avoid breaking things if you switch back and forth.
  - You should be able to search as with a single node cluster.
